l5kit/l5kit/tests/simulation/unroll_test.py

```diff
@@ -0,0 +1,103 @@
+from typing import Dict
+
+import numpy as np
+import pytest
+import torch
+
+from l5kit.data import ChunkedDataset, filter_agents_by_track_id, LocalDataManager
+from l5kit.dataset import EgoDataset
+from l5kit.rasterization import build_rasterizer
+from l5kit.simulation.unroll import SimulationConfig, SimulationLoop, SimulationDataset
+
+
+@pytest.fixture(scope="function")
+def ego_dataset(cfg: dict, dmg: LocalDataManager, zarr_cat_dataset: ChunkedDataset) -> EgoDataset:
+    rasterizer = build_rasterizer(cfg, dmg)
+    return EgoDataset(cfg, zarr_cat_dataset, rasterizer)
+
+
+class MockModel(torch.nn.Module):
+    def __init__(self, advance_x: float = 0.0):
+        super(MockModel, self).__init__()
+        self.advance_x = advance_x
+
+    def forward(self, x: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
+        centroids = x["centroid"]
+        bs = len(centroids)
+
+        positions = torch.zeros(bs, 12, 2, device=centroids.device)
+        positions[..., 0] = self.advance_x
+
+        yaws = torch.zeros(bs, 12, device=centroids.device)
+
+        return {"positions": positions, "yaws": yaws}
```

Comment:
Looks like it is repeated here, the `MockModel`.

Reply:
this will stau